include/pistache/endpoint.h

```diff
@@ -21,13 +21,15 @@ class Endpoint {
         Options& threads(int val);
         Options& flags(Flags<Tcp::Options> flags);
         Options& backlog(int val);
-        Options& maxPayload(size_t val);
```

Comment:
For now, please keep the old method as well.  Have it just call the new method (inline is fine), but give the old method the "__deprecated__" attribute, as documented:

https://gcc.gnu.org/onlinedocs/gcc-7.3.0/gcc/Common-Function-Attributes.html#Common-Function-Attributes